src/date-time/format-fuzzy-date.js

```diff
@@ -0,0 +1,85 @@
+import processPattern from '../util/process-pattern.js';
+
+const msPerDay = 86400000;
+
+const thirtySeconds = 30000;
+const ninetySeconds = 90000;
+
+const oneMinute = 60000;
+const twoAndAHalfMinutes = 150000;
+const fortyFiveMinutes = 2700000;
+
+const oneHour = 3600000;
+const oneHourThirtyMinutes = 5400000;
+const twoHoursThirtyMinutes = 9000000;
+
+export default function FuzzyDateFormatter(localeData, locale, timeZone) {
+	this.format = localeData.date.formats.fuzzyFormats;
+	this.locale = locale;
+	this.timeZone = timeZone ||  new Intl.DateTimeFormat().resolvedOptions().timeZone;
+}
+
+/*
+* Returns the fuzzy date string as determined by difference between now and date
+*/
+FuzzyDateFormatter.prototype.getDateString = function(inputDate, nowDate)  {
+
+	const referenceDate = nowDate || new Date();
+	const calendarDateDiff = (referenceDate - inputDate) / msPerDay;
+	const calendarDaysFloored = Math.floor(calendarDateDiff);
+	const sameYear = referenceDate.getYear() === inputDate.getYear();
+	const configuredTimeZoneHours = parseInt(new Intl.DateTimeFormat(this.locale, {hour:'numeric', hour12:false, timeZone: this.timeZone}).format(referenceDate).split(':'));
```

Comment:
new Intl.DateTimeFormat('en-us', {hour:'numeric', hour12:false}).format(new Date()) === 09 in firefox and chrome
new Intl.DateTimeFormat('en-us', {hour:'numeric', hour12:false}).format(new Date()) === 9:00 in edge/ie